TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -49,19 +52,73 @@ private SelectClause CreateSelectClause(CodeElementsParser.Sql_selectClauseConte
             SyntaxProperty<SelectionModifier> selectionModifier = null;
             if (context.SQL_ALL() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All, ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All,
+                    ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
             }
             else if (context.SQL_DISTINCT() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct, ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct,
+                    ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
             }
             if (context.star() != null)
             {
                 return new SelectClause(selectionModifier, new StarSelection());
             }
 
+            if (context.selections() != null)
+            {
+                var dotStarSelections = new List<DotStarSelection>();
+                int l = context.selections().ChildCount;
+                int i = 0;
+                foreach (var selection in context.selections().selection())
+                {
+                    dotStarSelections.Add(CreateDotStarSelection(selection));
+                }
+                return new SelectClause(selectionModifier, dotStarSelections);
+            }
+
             return null;
+        }
+
+        public DotStarSelection CreateDotStarSelection(CodeElementsParser.SelectionContext context)
+        {
+            if (context.dotStarSelection() != null)
+            {
+                if ((context.dotStarSelection().tableOrViewOrCorrelationName().Name != null))
```

Comment:
Put `context.dotStarSelection().tableOrViewOrCorrelationName()` into a variable to avoid calling theses methods multiple times. Also when using the `as` operator you don't have to chekch for null before. A null value can be cast into any reference type so this can be simplified as:

```csharp
if (context.dotStarSelection()?.tableOrViewOrCorrelationName() != null)
{
    var tableOrViewOrCorrelationName = context.dotStarSelection().tableOrViewOrCorrelationName();
    Token name = tableOrViewOrCorrelationName.Name as Token;
    Token schemaName = tableOrViewOrCorrelationName.SchemaName as Token;
    Token dbms = tableOrViewOrCorrelationName.Name.DBMS as Token;
}
```